src/CoreBundle/Resources/public/javascript/table.js

```diff
@@ -181,52 +183,150 @@ $(document).ready(function () {
+
+        select2SearchField.select();
+
+        $(select2SearchField).off('input');
+
+        if ('1' === this.dataset.keyUpListenerAttached) {
+            return;
+        }
+        select2SearchField.addEventListener('keypress', (e) => {
+            /* Preserve clear button */
+            const select2SelectionClear = selectSelectionRendered.querySelector('.select2-selection__clear');
+            selectSelectionRendered.title = select2SearchField.value;
+            selectSelectionRendered.textContent = select2SearchField.value;
+            if (null !== select2SelectionClear) {
+                selectSelectionRendered.appendChild(select2SelectionClear);
+            }
+        });
+
+        this.dataset.keyUpListenerAttached = '1';
     });
 
-    function switchRecord(id) {
+    function hasEventData(e) {
+        const argOriginalSelect2Event = e.params.args.originalSelect2Event;
+
+        if ('undefined' === typeof argOriginalSelect2Event) {
+            return false;
+        }
+
+        const data = argOriginalSelect2Event.data;
+
+        return 'undefined' !== typeof data;
+    }
+
+    function getDataFromEvent(e) {
+        return  e.params.args.originalSelect2Event.data;
+    }
+
+    function isSelected(originalSelect2EventData) {
+        if (true === originalSelect2EventData.newOption) {
+            return false;
+        }
+
+        if ('string' !== typeof originalSelect2EventData.id) {
+            return false;
+        }
+
+        if (originalSelect2EventData.id.trim() === originalSelect2EventData.text.trim()) {
+            return false;
+        }
+
+        return '' !== originalSelect2EventData.id.trim();
+    }
+
+    function isSearchTermEmpty(originalSelect2EventData) {
+        return '' === originalSelect2EventData.text.trim();
+    }
+
+    function getValueFromSearchField(searchField) {
+        if (null === searchField) {
+            return '';
+        }
+
+        return searchField.value;
+    }
+
+    function getSearchTerm(originalSelect2EventData) {
+        return originalSelect2EventData.text.trim();
+    }
+
+    function searchFieldContainsValue(searchField) {
+        if (null === searchField) {
+            return false;
+        }
+
+        const value = searchField.value;
+        if ('' === value) {
+            return false;
+        }
+
+        return value;
+    }
+
+    function switchRecord(id, searchTerm = '') {
         if ('' !== id) {
-            var url = searchLookup.data('record-url') + '&id=' + id;
-            top.document.location.href = url;
+            const searchTermPart = '' !== searchTerm ? `&_search_word=${searchTerm}` : '';
+            /* Here the URL will be set to make the browser load the given detail page's URL. The search form will not
+             * be submitted. Therefore the search term stored in the session will not be changed.
+             * In some cases it may be desirable to change the session stored search term to the selection value of the
```

Comment:
Where is this then decided (if the session search term is changed)?

Reply:
It cannot be decided by JavaScript code. Whatever goes to the session must be desided by the controller logic.

Some background on why I think it advantageous to submit the selectes option here:
There a two ways a User can interact with the select2 widget:

1. The user enters a search term and presses enter (to submit)
2. The user enters a search term and selects one of the options suggested by the select2 widget

Regarding 1: A search form will be submitted by JS and the search term can be stored in the session
Regarding 2: The browsers URL will be changed to that of the detail page of the selected option. In this case the search form will not be submitted. The session stored search term will remain as is.

The Problem:
Say on the product list page a user types "bracelet' into the search input field and presses the enter key. The search term "bracelet" will be stored in the session (and a result list will be displayed).
Next the user types "ocean" into the search input field and selects one of the suggested options. The user's browser will load the detail page of the selected option. The search term will remain "bracelet".
Next the user navigates back to the product list page. The search term "bracelet" will be prefilled in the search input field (from the session).
I think the expected behaviour would be to display either "ocean" (which is what the user last typed) or the complete title of the sleected option (i.e. "Armband Ocean - 17 cm"). 
To make this possible the JavaScrip code adds the complete title of the selected option as a GET parameter to the detail page's URL with the key `_search_word`. subsequently, the controller that handles the resulting request can decide whether or not to store the parameters value to the session. 
As yet there is no function triggered by submitting a `_search_word` parameter by JavaScript. I would like to suggest to cahnge the detail page controller accordingly. I will open a new ticket to propose that.

Reply:
If this is not used I would (only) do this code change within the new ticket.

Reply:
Done.